src/resolve/resolvconf-compat.c

```diff
@@ -51,7 +51,7 @@ static int parse_nameserver(const char *string) {
                 if (r == 0)
                         break;
 
-                if (strv_push(&arg_set_dns, word) < 0)
+                if (strv_push(&arg_set_dns, TAKE_PTR(word)) < 0)
```

Comment:
`TAKE_PTR()` here is not safe. When `strv_push()` fails, the variable `word` is not freed.